manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/MatchTest.scala

```diff
@@ -19,347 +19,305 @@
+        query(
+          """MATCH (martin:Person {name:"Martin Sheen"} ),
+            |      (oliver:Person {name:"Oliver Stone"}),
+            |      p = shortestPath( (martin)-[*..15]-(oliver) )
+            |RETURN p""", assertShortestPathLength) {
+          resultTable()
+          p(
+            """This means: find a single shortest path between two nodes, as long as the path is max 15 relationships long. Inside of the parentheses
+              | you define a single link of a path -- the starting node, the connecting relationship and the end node. Characteristics describing the relationship
+              | like relationship type, max hops and direction are all used when finding the shortest path. You can also mark the path as optional.""")
+        }
+      }
+      section("All shortest paths") {
+        p("Finds all the shortest paths between two nodes.")
+        query("""match (martin:Person {name:"Martin Sheen"} ),
+                |      (michael:Person {name:"Michael Douglas"}),
+                |      p = allShortestPaths( (martin)-[*]-(michael) ) return p""", assertAllShortestPaths) {
+          p("Finds the two shortest paths between Martin and Michael.")
+          resultTable()
+        }
+      }
+    }
+    section("Get node or relationship by id") {
+      section("Node by id", "match-node-by-id") {
+        p("Search for nodes by id can be done with the 'id' function in a predicate.")
+        note {
+          p("""Neo4j reuses its internal ids when nodes and relationships are deleted.
+              |This means that applications using, and relying on internal Neo4j ids, are brittle or at risk of making mistakes.
+              |Rather use application generated ids.""")
+        }
+        query("match (n) where id(n) = 0 return n", assertHasNodes(0)) {
+          p("The corresponding node is returned.")
+          resultTable()
+        }
+      }
+      section("Relationship by id") {
+        p("""
+            |Search for relationships by id can be done with the 'id' function in a predicate.
+            |
+            |This is not recommended practice. See <<match-node-by-id>> for more information on the use of Neo4j ids.
+            |""")
+        query("match ()-[r]->() where id(r) = 0 return r", assertHasRelationship0) {
+          p("The relationship with id +0+ is returned.")
+          resultTable()
+        }
+      }
+      section("Multiple nodes by id") {
+        p("Multiple nodes are selected by specifying them in an IN clause.")
+        query("match (n) where id(n) in [0,3,5] return n", assertHasNodes(0, 3, 5)) {
+          p("This returns the nodes listed in the `IN` expression.")
+          resultTable()
+        }
+      }
+    }
+  }.build()
+
+  private def assertAllNodesReturned = ResultAndDbAssertions((p, db) => {
+    val tx = db.beginTx()
+    try {
+      val allNodes: List[Node] = GlobalGraphOperations.at(db).getAllNodes.asScala.toList
```

Comment:
I guess it doesn't matter much, but I recall @pontusmelke saying `toList` is much slower than `toVector`. How large is this db?